specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-11-01/keysManagedHsm.json

```diff
@@ -0,0 +1,666 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2022-11-01",
+    "title": "KeyVaultManagementClient",
+    "description": "The Azure management API provides a RESTful set of web services that interact with Azure Key Vault."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/managedHSMs/{name}/keys/{keyName}": {
+      "put": {
+        "tags": [
+          "MHSMKeys"
+        ],
+        "operationId": "MHSMKeys_CreateIfNotExist",
+        "description": "Creates the first version of a new key if it does not exist. If it already exists, then the existing key is returned without any write operations being performed. This API does not create subsequent versions, and does not update existing keys.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group which contains the specified managed HSM."
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the managed HSM which contains the key to be created."
+          },
```

Comment:
Refactoring suggestion that again helps with reviewing and things being obviously consistent - you already define the hsm name parameter in managedHsm.json - why not put it in the "Parameters" section of that json, and use $ref to it from here?

Reply:
Done.